src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -307,6 +297,26 @@ private async Task ProcessSends()
             }
         }
 
+        private void FireConnectionClosed()
+        {
+            // Guard against scheduling this multiple times
+            if (_connectionClosed)
```

Comment:
I see why you added this bool for the libuv transport, but it seems completely unnecessary in the socket transport. FireConnectionClosed can only ever be called once.

Reply:
Yes that is true. I noticed that when I was copying code.